<?xml version="1.0"?>

<launch>

  <arg name="target_name"         value="boat_landing_platform"/>
  <arg name="is_serial_port" default="/dev/ttyUSB0"/>
  <!-- <arg name="need_ref_lla" value="true"/> -->
  <arg name="base_serial_port" default="/dev/ttyACM0"/>
  <arg name="compass_serial_port" default="/dev/ttyACM1"/>
  <arg name="base_host" default="192.168.0.143"/>
  <arg name="rover_host" default="192.168.0.84"/>

  <group ns="boat">
    <!-- Inertial Sense Node -->
    <rosparam subst_value="True">
      inertial_sense: { port: "/dev/ttyUSB0",
                            baud: 921600,
                            navigation_dt_ms: 10,
                            stream_INS: false,
                            stream_IMU: true,
                            stream_GPS: false,
                            stream_GPS_info: false,
                            stream_baro: true,
                            stream_mag: true,
                            stream_preint_IMU: false,
                            inclination: 1.14878541071,
                            declination: 0.20007290992,
                            dynamic_model: 5,
                            INS_rpy: [1.5708, 0 0],
                            INS_xyz: [1, 0, 0],
                            GPS_ant_xyz: [1, 1, 1],
                            GPS_ref_lla: [40.25, -111.67, 1556.59],
                            serial1_baud_rate: 115200,
                            NMEA_rate: 200,
                            NMEA_configuration: 0x0F,
                            NMEA_ports: 0x02
                          }

      </rosparam>

    <node name="inertial_sense" type="inertial_sense_node" pkg="inertial_sense" output="screen"/>
	  
      <!-- GPS Node-->
      <node name="f9p" pkg="ublox" type="ublox_ros">
        <param name="serial_port" value="$(arg base_serial_port)"/>
        <param name="local_host" value="$(arg base_host)"/>
        <param name="local_port" value="16140"/>
        <param name="rover_host" value="$(arg rover_host)"/>
        <param name="rover_port" value="16140"/>
        <param name="local_host2" value="$(arg base_host)"/>
        <param name="local_port2" value="16145"/>
        <param name="rover_host2" value="$(arg base_host)"/>
        <param name="rover_port2" value="16150"/>
        <rosparam command="load" file="$(find ublox)/params/ublox.yaml"/>
        <param name="rover_quantity" value="2"/>
	<param name="base_type" value="moving"/>
	<param name="message_rate" value="5"/>
      </node>

      <group ns="compass">
        <node name="f9p" pkg="ublox" type="ublox_ros">
          <param name="serial_port" value="$(arg compass_serial_port)"/>
          <param name="local_host" value="$(arg base_host)"/>
          <param name="local_port" value="16150"/>
          <param name="base_host" value="$(arg base_host)"/>
          <param name="base_port" value="16145"/>
          <rosparam command="load" file="$(find ublox)/params/ublox.yaml"/>
          <param name="rover_quantity" value="0"/>
          <param name="base_type" value="moving"/>
          <param name="message_rate" value="5"/>
        </node>
      </group>

      <!-- Estimator -->
    <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
      <remap from="imu" to="imu"/> <!--need to figure out the right topic here-->
      <remap from="baro" to="dummy_topic"/>
      <remap from="ublox_gnss" to="PosVelEcef"/>
      <remap from="ublox_relpos" to="compass/RelPos"/>
      <remap from="common_ref_lla" to="/ref_lla"/>
    </node>

    <node name="is_throttler" type="throttle" pkg="topic_tools" args="messages /imu 10 /imu_throttled"/>

  </group>

  <node pkg="rosbag" type="record" name="record" output="screen" args="-O ../data/base --all"/>

</launch>
